testbed_api/camera.py

```diff
@@ -7,30 +10,58 @@ class VideoCamera:
 
     def __init__(self, index: int = 0):
         self._capture = cv2.VideoCapture(index)
+        self._capture.set(cv2.CAP_PROP_FOURCC, cv2.VideoWriter_fourcc('X', 'V', 'I', 'D'))
+        self._prev_frame_time = 0
+        self._new_frame_time = 0
+        
+        self._frames = deque()
+        self._thread = threading.Thread(target = self.read_frames)
+        self._thread.start()
 
+    
     def __del__(self):
         self._capture.release()
+        self._thread.join()
 
+    
     def get_frame(self):
-        success, image = self._capture.read()
-        assert success
-        retval, jpeg = cv2.imencode('.jpg', image)
-        assert retval
-        return jpeg.tobytes()
+        while (not self._frames):
+            time.sleep(0.005)
+        return self._frames.pop()
```

Comment:
Getting a single frame now consists of quickly retrieving an item from the queue